tools/rpkg/R/register.R

```diff
@@ -36,3 +36,34 @@ duckdb_unregister <- function(conn, name) {
   invisible(TRUE)
 }
 
+#' Register an Arrow data source as a virtual table
+#'
+#' `duckdb_register_arrow()` registers an Arrow data source as a virtual table (view)
+#'  in a DuckDB connection.
+#'  No data is copied.
+#'
+#' `duckdb_unregister_arrow()` unregisters a previously registered data frame.
+#' @param conn A DuckDB connection, created by `dbConnect()`.
+#' @param name The name for the virtual table that is registered or unregistered
+#' @param arrow_scannable A scannable Arrow-object
+#' @return These functions are called for their side effect.
+#' @export
+duckdb_register_arrow <- function(conn, name, arrow_scannable) {
+  stopifnot(dbIsValid(conn))
+
+    # create some R functions to pass to c-land
+    export_fun <- function(arrow_scannable, stream_ptr) {
+        record_batch_reader <- arrow::Scanner$create(arrow_scannable)$ToRecordBatchReader()
+        arrow:::ExportRecordBatchReader(record_batch_reader, stream_ptr)
```

Comment:
To get around the `:::` issue for now, how about this?

```suggestion
        # This function is not yet exported from {arrow}
        # Remove this workaround when it is exported
        get("ExportRecordBatchReader", envir = asNamespace("arrow"))(record_batch_reader, stream_ptr)
```

Reply:
Why not use `getFromNamespace` then? `:::` triggers a CRAN check note, which I would like to avoid. Not sure what `get` or `getFromNamespace` will do.

Reply:
This doesn't cause a note, which is why I suggested it.